### PR TITLE
Add support for configuring overlay2 on peons

### PR DESCRIPTION
**Pull request by cevich**
[--args=docker_cli/run_flags]

### CODE REVIEW

**Comment by cevich** [comment]
Shot-in-the-dark...no idea if this is all changes needed...relying on CI to break to know one way or t'other.

File: jobs/basic/inventory/peons [hunk 1]
```diff
@@ -5,6 +5,8 @@
 
 rhel7-4-docker
 rhelatomic7-4-docker-latest
+rhelatomic7-4-docker-latest-overlay2
 
 fedora26-docker-latest
+fedora26-docker-latest-overlay2
```
Comment (edsantiago) [inline jobs/basic/inventory/peons]: Probably unnecessary: f26 already defaults to overlay2

File: kommandir/inventory/peons [hunk 2]
```diff
@@ -16,6 +17,7 @@ rhel7-2-docker-latest
 
 rhelatomic7-4-docker
 rhelatomic7-4-docker-latest
+rhelatomic7-4-docker-latest-overlay2
```
Comment (edsantiago) [inline kommandir/inventory/peons]: I don't know if this is possible: by the time rhelatomic comes up, docker has started and has configured its storage.

File: kommandir/inventory/peons [hunk 3]
```diff
@@ -25,6 +27,7 @@ rhelatomic7-2-docker-latest
 
 fedora26-docker-latest
 fedoratomic26-docker-latest
+fedoratomic26-docker-latest-overlay2
```
Comment (edsantiago) [inline kommandir/inventory/peons]: See above re: fedora26 default

**Comment by cevich** [comment]
[(1) [CN] Add overlay/overlay2 VMs to 'full' CI job](https://trello.com/c/miZInSgk/345-1-cn-add-overlay-overlay2-vms-to-full-ci-job)

**Comment by cevich** [comment]
> not a complete review; just commenting on some stuff that may be unnecessary

Thanks Ed!  I really appreciate you're feedback, esp. since you said ADEPT-tea burns straight through the bottom of your fancy pottery mug 😀 

>  I don't know if this is possible: by the time rhelatomic comes up, docker has started and has configured its storage.

Tl;DR; The way this is intended to work (documented today, literally) is:

* Peons are joined to a "distro" group via [their ``peon_groups`` variable](http://autotest-docker-enabled-product-testing.readthedocs.io/en/latest/#peon-groups)
* Each group hauls in a bunch of related variables, so e.g. the ``rhel`` group would suck in:
``./kommandir/inventory/group_vars/rhel/*.yml``
* An important group is the kind of cloud to use, e.g. ``openstack``, so [bringing in ``cloud_provisioning_command``](http://autotest-docker-enabled-product-testing.readthedocs.io/en/latest/#cloud-provisioning-command), in this case from ``./kommandir/inventory/group_vars/openstack/*.yml``. 
* Additionally, I added a file with cloud-specific details to each distro. group, e.g.
  ``kommandir/inventory/group_vars/rhel/openstack.yml`` (file name isn't important, only contents)
* In that file the cloud-specific variable ``peon_userdata`` is set, to be consumed by the ``cloud_provisioning_command`` (mentioned above).
* In the ``openstack`` case, it sets:  ``peon_userdata: '{{ hostvars.kommandir.workspace ~ "/roles/peon_created/files/nodocker_userdata.yml" if "openstack" in group_names else omit }}'``
* The ``nodocker_userdata.yml`` file causes cloud-init to disable the docker service, and kill ``/var/log/docker*`` before the service can start (I got eng. approval on this).
* This allows the playbooks to apply roles which configure docker-storage-setup and items like overlay2 on atomic peons.

Whew! Okay, I'll check the job to see how badly it blew up...

**Comment by cevich** [comment]
mmm.  The ``rhelatomic7-4-docker-latest-overlay2`` peon never got provisioned...

**Comment by cevich** [comment]
OIC what's happening, for some reason at execution time, it's grabbing an internal copy of the inventory instead of the one from the PR.

**Comment by cevich** [comment]
Fixed: Moved PR testing job details from internal repo. to here.

**Comment by cevich** [comment]
Hmmm...I seen that error before, thought I fixed it....

**Comment by cevich** [comment]
--rebase && --autosquash all tests should pass now
